File: calculator/src/LocalStorage.ts
import {Category, PlayerFaction} from "./Models";

const FACTION_KEY = "player-faction"

const SELECTED_ITEM_KEY_PREFIX = "selected-item"

export function setPlayerFaction(faction: PlayerFaction) {
    localStorage.setItem(FACTION_KEY, faction)
}

export function getPlayerFaction(): PlayerFaction {
    return localStorage.getItem(FACTION_KEY) as PlayerFaction | "colonial"
}

export function getSavedSelectedItemName(category: Category): string {
    return localStorage.getItem(selectedItemKey(category)) as string | ""
}

export function setSavedSelectedItemName(category: Category, itemName: string) {
    localStorage.setItem(selectedItemKey(category), itemName)
}

function selectedItemKey(category: Category): string {
    return `${SELECTED_ITEM_KEY_PREFIX}-${category}`
}

File: calculator/src/mpf-calculatorMain/kotlin/LocalStorage.kt
import de.chasenet.foxhole.model.Faction
import de.chasenet.foxhole.model.ItemCategory
import kotlinx.browser.localStorage

private const val PLAYER_FACTION = "player-faction"

private fun selectedItemKey(category: ItemCategory) = "selected-item-${category.name}"

fun getSavedSelectedItem(category: ItemCategory): String = localStorage.getItem(selectedItemKey(category)) ?: ""

fun saveSelectedItem(
    category: ItemCategory,
    itemName: String,
) {
    localStorage.setItem(selectedItemKey(category), itemName)
}

fun getPlayerFaction(): Faction = localStorage.getItem(PLAYER_FACTION)?.let(Faction::valueOf) ?: Faction.colonial

fun savePlayerFaction(faction: Faction) {
    localStorage.setItem(PLAYER_FACTION, faction.name)
}
